library/system/src/lib/yast2/ui_plugin_info.rb

```diff
@@ -0,0 +1,162 @@
+  # the current process) from its /proc/self/maps file.
+  #
+  # You can also get information for a different process by specifying its
+  # /proc/$pid/maps file.
+  #
+  # For testing, a fixed file can be used.
+  #
+  # The information stored in this class is only a snapshot in the life time of
+  # the process. As new shared libs are loaded (e.g. plug-ins are loaded with
+  # dlopen()), this information may become outdated. In that case, simply let
+  # the old instance go out of scope and create a new one.
+  #
+  # More information: https://github.com/yast/yast-yast2/pull/1194
+  #
+  class UiPluginInfo < SharedLibInfo
+    include Yast::Logger
+
+    # Constructor.
+    # @param maps_file [String] name of the maps file to use
+    #
+    def initialize(maps_file = "/proc/self/maps")
+      super
+      # Lazy init for those member variables
+      @ui_plugins = nil
+      @main_ui_plugin_complete = nil
+    end
+
+    # Find the UI plug-ins among the shared libs.
+    #
+    # @return [Array<String>] Complete paths of the UI plug-ins
+    #
+    def ui_plugins
+      @ui_plugins ||= shared_libs.select { |lib| lib =~ /yui\/libyui-/ }
+      log.info("UI plug-ins: #{@ui_plugins}")
+      @ui_plugins
+    end
+
+    # Return the short name of a UI plug-in, i.e. only the lib base name with
+    # any leading "libyui-" removed, i.e. something like "qt", "ncurses".
+    #
+    # @param ui_plugin [String] full name (with or without path) of the UI plug-in
+    # @return [String] corresponding short name
+    #
+    def short_name(ui_plugin)
+      name = SharedLibInfo.lib_basename(ui_plugin)
+      name&.gsub(/^libyui-/, "")
+    end
+
+    # Return the complete name (with path and SO number) of the main UI
+    # plug-in.  Several UI plug-ins might be loaded; the main plug-in is
+    # generally the one with the shortest lib base name ("qt", "qt-pkg",
+    # "qt-graph"; "ncurses", "ncurses-pkg").
+    #
+    # @return [String, nil] Complete name of the main UI plug-in
+    #
+    def main_ui_plugin_complete
+      return nil if ui_plugins.empty?
+
+      relevant_plugins = ui_plugins.reject { |p| p =~ /rest-api/ }
+      @main_ui_plugin_complete ||= relevant_plugins.min do |a, b|
+        SharedLibInfo.lib_basename(a).size <=> SharedLibInfo.lib_basename(b).size
+      end
+      @main_ui_plugin_complete
+    end
+
+    # Return the short name of the main UI plug-in, i.e. without path,
+    # "libyui-" prefix and SO number Several UI plug-ins might be loaded; the
+    # main plug-in is generally the one with the shortest lib base name ("qt",
+    # "qt-pkg", "qt-graph"; "ncurses", "ncurses-pkg").
+    #
+    # @return [String, nil] Short name of the main UI plug-in
+    #
+    def main_ui_plugin
+      name = SharedLibInfo.lib_basename(main_ui_plugin_complete)
+      name&.gsub!(/^libyui-/, "")
+    end
+
+    # Find the SO number of the UI main plug-in.
+    #
+    # @return [String, nil] SO number (e.g. "15.0.0")
+    #
+    def ui_so_number
```

Comment:
Maybe we could drop the `ui_` prefixes as the class is already called **Ui**PluginInfo.
You will be calling `UiPluginInfo.ui_so_number`, IMHO `UiPluginInfo.so_number` is shorter and still contains everything.

Reply:
This is intentional to avoid (minor) confusion with similar base class methods, albeit they tend to be _class_ methods. Still, I'd like to be explicit about that here.

Reply:
I see, I overlooked that the names already exist in the base class...